scripts/ingests/ingest_utils.py

```diff
@@ -1307,3 +1340,36 @@ def ingest_gaia_pms(db, sources, gaia_data, ref):
                           pms['pmdec'], pms['pmdec_error'],
                           refs)
 
+
+def ingest_spectrum_from_fits(db, source, spectrum_fits_file):
+    """
+    Ingests spectrum using data found in the header
+
+    Parameters
+    ----------
+    db
+    source
+    spectrum_fits_file
+
+    """
+    header = fits.getheader(spectrum_fits_file)
+    regime = header['SPECBAND']
+    if regime == 'opt':
+        regime = 'optical'
+    telescope = header['TELESCOP']
+    instrument = header['INSTRUME']
+    try:
+        mode = header['MODE']
+    except KeyError:
+        mode = None
+    obs_date = header['DATE-OBS']
+    doi = header['REFERENC']
+    data_header = fits.getheader(spectrum_fits_file, 1)
+    w_unit = data_header['TUNIT1']
+    flux_unit = data_header['TUNIT2']
+
+    reference_match = db.query(db.Publications.c.publication).filter(db.Publications.c.doi == doi).table()
+    reference = reference_match['publication'][0]
```

Comment:
would this line give an error if there is no doi?  I don't know too much about fits files, and if they come with doi's but if they don't then make we could do a try and except for the reference match  which could also maybe make sense for adding unpublished data ?

Reply:
You are absolutely correct! This script needs a ton of error catching. I will add a little bit and add some TODOs since I really want to get the data ingested ASAP.

Reply:
I've recorded this todo is issue #297.